lib/blocks.js

```diff
@@ -787,20 +782,32 @@ var blockStarts = [
             return 0;
         }
 
+        const nextColumn = measureNonspaceColumn(parser.nextLine);
+        if (nextColumn - parser.column >= CODE_INDENT || parser.column - nextColumn >= CODE_INDENT) {
+            // the delimiter row must be on the same indentation level as the header row
+            return 0;
+        }
+
+        parser.advanceNextNonspace();
+
         // check for a delimiter first since it's stricter than the header row
-        const nextLine = trimSpacesAfterPipe(parser.nextLine);
```

Comment:
`trimSpacesAfterPipe` was added as a fix for https://mattermost.atlassian.net/browse/MM-13516. Before the change I commented on above, we could only trim the trailing whitespace before passing the line into `reTableDelimiter`. With the whitespace handled separately, we can just use a regular old `.trim` to get rid of that whitespace